src/App.jsx

```diff
@@ -11,37 +11,42 @@ function App() {
   const contentRef = useRef(null);
   const [isStickyFooter, setIsStickyFooter] = useState(false);
   const headerRef = useRef(null);
-  const footerRef = useRef(null);
   const [headerHeight, setHeaderHeight] = useState(0);
-  const [footerHeight, setFooterHeight] = useState(0);
+  const [viewportHeight, setViewportHeight] = useState(window.innerHeight);
+
+  useEffect(() => {
+    //used for when you adjust the height of the window
+    const handleResize = () => {
+      setViewportHeight(window.innerHeight);
+    };
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
 
   useEffect(() => {
     const contentHeight = contentRef.current.getBoundingClientRect().height;
-    const viewportHeight = window.innerHeight;
-    // Determine if the footer should be sticky
-    // sets isStickerFooter to true if the content height is less than the viewport height, false otherwise
-    setIsStickyFooter(contentHeight < viewportHeight);
-  }, [location.pathname]); // Rerun this effect if the route changes
+    // 84 is the height of the footer across all screen sizes, using forwardRef caused
+    // the footer to always be sticky for some reason. This is a hacky fix for now
+    const workingHeight = contentHeight + headerHeight + 84;
```

Comment:
If the 84 gets changed then this will probably break. One strategy for this is to pull the variable out into a central location like a new file where it has

`const FOOTER_HEIGHT = 84` 
(the convention for naming constant values that never change throughout the life of the program is to have them all upper case with underscores)

And then you can import that into all the places that use it so that you can just change that one spot and it'll update all the places that use it. BUT Im guessing that the other file that uses this 84 value is a css file and so I dont know if you can import variables into that file so this strategy might not work. If thats not possible then just leave a comment above both places that use the 84 value saying if you change that value then you need to go to the other place and change it there as well.

Reply:
This is the only file where 84 is used. The footer just happened to end up at 84px high after we styled and added the logo. It remains that height on all screen sizes. But in the footer.css file, if we change the padding of .footer-container or height of #footer-logo, then the footer itself will change from 84px. i'll leave a comment in the css saying if we were to ever change those values, then we have to check what height that changed the footer to and then adjust it in app.jsx

Reply:
I'll leave a comment in app.jsx as well

Reply:
Cool! Can we also just lock the footer to that height explicitly rather than the height being determined by the logo size or padding?  I'm not really sure how this works

Reply:
yeah thats probably better than adjusting the size of the footer depending on what we want to do with the content of the footer. 

we can set the footer height to 84, and make the content work around that